src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -70,6 +83,59 @@ public VectorSearchConfig(Integer dimension, String metric) {
       }
     }
 
+    public record IndexingConfig(
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "List of allowed indexing fields",
+                type = SchemaType.ARRAY,
+                implementation = String.class)
+            @Nullable
+            List<String> allow,
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+            @Schema(
+                description = "List of denied indexing fields",
+                type = SchemaType.ARRAY,
+                implementation = String.class)
+            @Nullable
+            List<String> deny) {
+
+      public void validate() {
+        if (allow() != null && deny() != null) {
+          throw new JsonApiException(
+              ErrorCode.INVALID_INDEXING_USAGE,
+              ErrorCode.INVALID_INDEXING_USAGE.getMessage()
+                  + " - allow and deny cannot be used together");
```

Comment:
Nit: should also quotes (single or double?) around "allow" and "deny" to indicate they are properties.